tests/test-ajax.php

```diff
@@ -0,0 +1,128 @@
+			return;
+		}
+
+		// Check that the exception was thrown.
+		$response = json_decode( $this->_last_response );
+
+		$this->assertInternalType( 'object', $response );
+		$this->assertObjectHasAttribute( 'success', $response );
+		$this->assertTrue( $response->success );
+		$this->assertObjectHasAttribute( 'data', $response );
+	}
+
+	public function test_ajax_error_bad_nonce() {
+		$_POST['nonce']     = uniqid();
+		$_POST['action']    = 'stickytax_get_id_from_name';
+		$_POST['term_name'] = 'Uncategorized';
+		$_POST['term_type'] = 'category';
+
+		try {
+			$this->_handleAjax( 'stickytax_get_id_from_name' );
+		} catch ( WPAjaxDieContinueException $e ) {
+			return;
+		}
+
+		// Check that the exception was thrown.
+		$response = json_decode( $this->_last_response );
+
+		$this->assertInternalType( 'object', $response );
+		$this->assertObjectHasAttribute( 'success', $response );
+		$this->assertFalse( $response->success );
+		$this->assertObjectHasAttribute( 'data', $response );
+		$this->assertObjectHasAttribute( 'errcode', $response->data );
+		$this->assertEquals( $response->data->errcode, 'BAD_NONCE' );
+	}
+
+	public function test_ajax_error_no_term_name() {
+		$_POST['nonce']     = wp_create_nonce( 'sticky-tax' );
+		$_POST['action']    = 'stickytax_get_id_from_name';
+		$_POST['term_name'] = '';
+		$_POST['term_type'] = 'category';
+
+		try {
+			$this->_handleAjax( 'stickytax_get_id_from_name' );
+		} catch ( WPAjaxDieContinueException $e ) {
+			return;
+		}
+
+		// Check that the exception was thrown.
+		$response = json_decode( $this->_last_response );
+
+		$this->assertInternalType( 'object', $response );
+		$this->assertObjectHasAttribute( 'success', $response );
+		$this->assertFalse( $response->success );
+		$this->assertObjectHasAttribute( 'data', $response );
+		$this->assertObjectHasAttribute( 'errcode', $response->data );
+		$this->assertEquals( $response->data->errcode, 'NO_TERM_NAME' );
+	}
+
+	public function test_ajax_error_no_term_type() {
+		$_POST['nonce']     = wp_create_nonce( 'sticky-tax' );
+		$_POST['action']    = 'stickytax_get_id_from_name';
+		$_POST['term_name'] = 'Uncategorized';
+		$_POST['term_type'] = '';
+
+		try {
+			$this->_handleAjax( 'stickytax_get_id_from_name' );
+		} catch ( WPAjaxDieContinueException $e ) {
+			return;
+		}
+
+		// Check that the exception was thrown.
+		$response = json_decode( $this->_last_response );
+
+		$this->assertInternalType( 'object', $response );
+		$this->assertObjectHasAttribute( 'success', $response );
+		$this->assertFalse( $response->success );
+		$this->assertObjectHasAttribute( 'data', $response );
+		$this->assertObjectHasAttribute( 'errcode', $response->data );
+		$this->assertEquals( $response->data->errcode, 'NO_TERM_TYPE' );
+	}
+
+	public function test_ajax_error_no_term_found() {
+		$_POST['nonce']     = wp_create_nonce( 'sticky-tax' );
+		$_POST['action']    = 'stickytax_get_id_from_name';
+		$_POST['term_name'] = uniqid();
```

Comment:
I'm a huge fan of `uniqid()` in tests and use it regularly, but it may not be the best representation of a term name (especially since WordPress can be kinda messy in how it handles finding terms by names). Instead, I'd recommend using a hard-coded string, but adding a pre-condition assertion:

```php
$_POST['term_name'] = 'this-term-does-not-exist';

// Verify that the non-existent term is, in fact, non-existent.
$this->assertFalse(
  (bool) term_exists( $_POST['term_name'], 'category',
  'Category "' . $_POST['term_name'] . '" expected not to exist.'
);
```